florestad/src/json_rpc/blockchain.rs

```diff
@@ -190,12 +190,35 @@ impl RpcImpl {
     }
 
     // gettxoutproof
-    pub(super) fn get_tx_proof(&self, tx_id: Txid) -> Result<Vec<String>, RpcError> {
-        Ok(self
-            .wallet
-            .get_merkle_proof(&tx_id)
-            .ok_or(RpcError::TxNotFound)?
-            .0)
+    pub(super) async fn get_txout_proof(
+        &self,
+        tx_ids: &[Txid],
+        blockhash: Option<BlockHash>,
+    ) -> Result<Vec<Vec<String>>, RpcError> {
+        let targets = match blockhash {
+            Some(blockhash) => {
+                let ret: Vec<Txid> = self
+                    .get_block_inner(blockhash)
+                    .await?
+                    .txdata
+                    .iter()
+                    .map(|tx| tx.compute_txid())
+                    .filter(|tx| tx_ids.contains(&tx))
+                    .collect();
+                ret
+            }
+            None => tx_ids.into(),
+        };
+        Ok(targets
```

Comment:
The block hash param is a hint about what block this tx might be, so we can build a proof, even though we don't have a txindex to assist. If the block hash is present, you should try to get the merkle path from the block itself.